subprojects/core-api/src/main/java/org/gradle/api/artifacts/transform/TransformAction.java

```diff
@@ -16,20 +16,42 @@
 
 package org.gradle.api.artifacts.transform;
 
+import org.gradle.api.Action;
 import org.gradle.api.Incubating;
 
 import javax.inject.Inject;
 
 /**
  * Interface for artifact transform actions.
  *
- * <p>Implementations must provide a public constructor.</p>
+ * <p>
+ *     A transform action implementation is an abstract class implementing the {@link #transform(TransformOutputs)} method.
+ *     A minimal implementation may look like this:
+ * </p>
  *
- * <p>Implementations can receive parameters by using annotated abstract getter methods.</p>
+ * <pre class='autoTested'>
+ * import org.gradle.api.artifacts.transform.TransformParameters
```

Comment:
Why do we need this import? Shouldn't this be auto-imported along with all the other types used in the sample?

Also, we should make up our minds whether this is Java or Groovy code. If Groovy, we should drop the `public` modifiers, and use idiomatic Groovy in the code, and no semicolons at the end of the line. If it's Java, then we should add a semicolon here.

Reply:
I would use Java code for the samples for the implementation and Groovy code when calling methods from the DSL, like `registerTransform`. WDYT?

Regarding the import: We need it for https://github.com/gradle/gradle/issues/8683

Reply:
> I would use Java code for the samples for the implementation and Groovy code when calling methods from the DSL, like `registerTransform`. WDYT?

I think that's good.



> Regarding the import: We need it for #8683

Let's leave a comment about this in the code, so when #8683 is fixes someone bumping into this can remove it.

Reply:
I left a comment on https://github.com/gradle/gradle/issues/8683#issuecomment-494467094.